### PR TITLE
RISC-V: updated intrin_rvv071.hpp to work with modern toolchain 2.8.0

### PR DESCRIPTION
**Pull request by mshabunin**
- intrinsics implementation (071) reworked to use modern RVV intrinsics syntax
- cmake toolchain file (071) now allows selecting from predefined configurations

##### Build instruction:
* use Xuantie toolchain 2.8.0: https://github.com/T-head-Semi/xuantie-gnu-toolchain
*  configure:
    ```
    PATH=/opt/xuantie-2.8.0/bin:${PATH} \
    cmake -GNinja \
     -DCMAKE_BUILD_TYPE=Release \
     -DBUILD_SHARED_LIBS=OFF \
     -DWITH_OPENCL=OFF \
     -DCMAKE_TOOLCHAIN_FILE=/opencv/platforms/linux/riscv64-071-gcc.toolchain.cmake \
     -DCORE=C910 \
     /opencv
    ```
* build: `ninja`

**UPD:**
* Updated toolchain file - removed unrelated configurations and tested all configurations (build only)
* Prepared Docker image with Xuantie 2.8.0 toolchain: https://github.com/opencv-infrastructure/opencv-gha-dockerfile/pull/29

TODO (after merge):
* update weekly RISC-V build and instruction on wiki

### CODE REVIEW

File: platforms/linux/riscv64-071-gcc.toolchain.cmake [hunk 1]
```diff
@@ -4,5 +4,57 @@ set(CMAKE_SYSTEM_PROCESSOR riscv64)
...
+  set(CMAKE_CXX_FLAGS_INIT "-march=rv64imafdc_zihintpause_zfh_zba_zbb_zbc_zbs_xtheadc -mabi=lp64d")
+endif()
+
+if(_enable_vector)
+  set(CMAKE_C_FLAGS_INIT "${CMAKE_C_FLAGS_INIT} -D__riscv_vector_071 -mrvv-vector-bits=128")
+  set(CMAKE_CXX_FLAGS_INIT "${CMAKE_CXX_FLAGS_INIT} -D__riscv_vector_071 -mrvv-vector-bits=128")
+endif()
+
+# set(CMAKE_CXX_FLAGS_INIT "-march=rv64gcv -mabi=lp64d -D__riscv_vector_071")
+# set(CMAKE_C_FLAGS_INIT "-march=rv64gcv -mabi=lp64d -D__riscv_vector_071")
```
Comment (opencv-alalek) [inline platforms/linux/riscv64-071-gcc.toolchain.cmake:55]: Do we need this commented code?

File: platforms/linux/riscv64-071-gcc.toolchain.cmake [hunk 2]
```diff
@@ -4,5 +4,57 @@ set(CMAKE_SYSTEM_PROCESSOR riscv64)
...
+
+if(ENABLE_GCOV)
+    set(CMAKE_CXX_FLAGS_INIT "${CMAKE_CXX_FLAGS_INIT} -fprofile-arcs -ftest-coverage")
+    set(CMAKE_C_FLAGS_INIT "${CMAKE_C_FLAGS_INIT} -fprofile-arcs -ftest-coverage")
```
Comment (opencv-alalek) [inline platforms/linux/riscv64-071-gcc.toolchain.cmake:59]: 2 spaces indentation is preferable in CMake

File: platforms/linux/riscv64-071-gcc.toolchain.cmake (hunk 1 above)
Comment (mshabunin) [reply platforms/linux/riscv64-071-gcc.toolchain.cmake]: No, I've removed it.

File: platforms/linux/riscv64-071-gcc.toolchain.cmake (hunk 2 above)
Comment (mshabunin) [reply platforms/linux/riscv64-071-gcc.toolchain.cmake]: Fixed.

**Review by opencv-alalek — APPROVED** [review]

**Pull request merged**